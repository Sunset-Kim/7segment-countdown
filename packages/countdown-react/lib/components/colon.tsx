import { CommonProps } from "../types";

export interface ColonProps extends CommonProps {
  colonGap?: number;
}

export const Colon: React.FC<ColonProps> = ({
  color = "dodgerblue",
  size = 100,
  colonGap = 60,
}) => {
  return (
    <svg width={size * 0.2} height={size * 2} viewBox="0 0 20 200" role="img">
      <circle cx={10} cy={100 - colonGap / 2} r={5} fill={color} />
      <circle cx={10} cy={100 + colonGap / 2} r={5} fill={color} />
    </svg>
  );
};
