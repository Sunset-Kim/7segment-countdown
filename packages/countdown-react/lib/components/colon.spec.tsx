import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import { Colon } from "./colon";

describe("Colon 컴포넌트", () => {
  it("기본 프롭스로 렌더링된 스냅샷과 일치해야 함", () => {
    const { container } = render(<Colon />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("커스텀 프롭스로 렌더링된 스냅샷과 일치해야 함", () => {
    const { container } = render(
      <Colon color="red" size={200} colonGap={80} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
