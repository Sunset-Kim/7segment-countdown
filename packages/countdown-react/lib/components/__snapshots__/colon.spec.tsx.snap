// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Colon 컴포넌트 > 기본 프롭스로 렌더링된 스냅샷과 일치해야 함 1`] = `
<svg
  height="200"
  role="img"
  viewBox="0 0 20 200"
  width="20"
>
  <circle
    cx="10"
    cy="70"
    fill="dodgerblue"
    r="5"
  />
  <circle
    cx="10"
    cy="130"
    fill="dodgerblue"
    r="5"
  />
</svg>
`;

exports[`Colon 컴포넌트 > 커스텀 프롭스로 렌더링된 스냅샷과 일치해야 함 1`] = `
<svg
  height="400"
  role="img"
  viewBox="0 0 20 200"
  width="40"
>
  <circle
    cx="10"
    cy="60"
    fill="red"
    r="5"
  />
  <circle
    cx="10"
    cy="140"
    fill="red"
    r="5"
  />
</svg>
`;
